FenwickTree/RangeUpdatePointQueries.java

```diff
@@ -0,0 +1,87 @@
+public class RangeUpdatePointQueries {
+	
+	public static void main(String[] args) {
+		RangeUpdatePointQueries d=new RangeUpdatePointQueries();
+		//input array
+		int[] inArr={0,0,0,0,0,0};
+		//Store the fenwick tree
+		int[] feniTree=new int[inArr.length+1];
+		//Update range using method 2
+		d.updateRangeMeth2(inArr, 2, 4, 10);
+		//Point query method 2
+		System.out.println(d.getPointMeth2(inArr, 3));
```

Comment:
Can you explain what the difference between getPoint and getPointMeth2? I noticed that getPoint has a logarithmic time complexity while getPointMeth2 has a linear complexity. getPointMeth2 seems impractical if it's to be so slow.

Reply:
updateRangeMeth2 and getPointMeth2- Its complexity is linear. It updates the input array with update values. If we want to update position 2 to 4 of input array with 10, it adds inArr[2]+=10 and inArr[5]+=10. When we want the point value of lets say position 3. We add all the values from position 0 to 3. That returns value 10. If we want point value for position 5. We add all the values from position 0 to 5. That returns value 0.